### ISSUE TITLE
Re-evaluate checked in generated sources

### ISSUE THREAD

**Issue by jannic** [rp-rs/rp2040-pac#78]
The generated sources in `src/` cause a lot of noise in merge requests.

Evaluate if there is a better way. As @ithinuel wrote in the matrix room: "I'm starting to wonder if you shan't remove the src from the main branch and rather have a pipeline run and generate a parallel branch with the generated code ala gh-pages."

Disadvantages of a separate branch could be:
- Increased workflow complexity
- Repo layout less obvious, especially for casual observers and newcomers
- Reviewers can't easily see the effect of changes on the resulting crate

**Comment by AkiyukiOkayasu** [comment]
I agree that it is difficult to review, but it is easier for beginners to understand if the source is in the main branch.  

I think it is the regression caused by svd2rust or svdtool updates that we should be worried about.
I have not fully researched this yet, but what about including rp2040 in [svd2rust-regress](https://github.com/rust-embedded/svd2rust/tree/master/ci/svd2rust-regress)?

**Comment by ithinuel** [comment]
We could keep the branch with the generated sources as the visible branch but ask for contributions to be made on a dev branch.

This'd keep the best of both worlds for the cost of _slightly_ increased complexity for contributors.
